src/AWS.Deploy.Recipes/CdkTemplates/AspNetAppEcsFargate/Generated/RecipeConstruct.cs

```diff
@@ -0,0 +1,342 @@
+using System;
+using System.Collections.Generic;
+
+using Amazon.CDK;
+using Amazon.CDK.AWS.ApplicationAutoScaling;
+using Amazon.CDK.AWS.EC2;
+using Amazon.CDK.AWS.ECS;
+using Amazon.CDK.AWS.ElasticLoadBalancingV2;
+using Amazon.CDK.AWS.IAM;
+using AWS.Deploy.Recipes.CDK.Common;
+using AspNetAppEcsFargate.Configurations;
+using Amazon.CDK.AWS.ECR;
+
+
+// This is a generated file from the original deployment recipe. It is recommended to not modify this file in order
+// to allow easy updates to the file when the original recipe that this project was created from has updates.
+// To customize the CDK constructs created in this file you should use the AppStack.CustomizeCDKProps() method.
+
+namespace AspNetAppEcsFargate
+{
+    using static AWS.Deploy.Recipes.CDK.Common.CDKRecipeCustomizer<RecipeConstruct>;
+
+
+    public class RecipeConstruct : Construct
+    {
+        public IVpc? AppVpc { get; private set; }
+        public ICluster? EcsCluster { get; private set; }
+        public IRole? AppIAMTaskRole { get; private set; }
+        public TaskDefinition? AppTaskDefinition { get; private set; }
+        public IRepository? EcrRepository { get; private set; }
+        public ContainerDefinition? AppContainerDefinition { get; private set; }
+        public SecurityGroup? WebAccessSecurityGroup { get; private set; }
+        public IList<ISecurityGroup>? EcsServiceSecurityGroups { get; private set; }
+        public FargateService? AppFargateService { get; private set; }
+        public IApplicationLoadBalancer? ServiceLoadBalancer { get; private set; }
+        public IApplicationListener? LoadBalancerListener { get; private set; }
+        public ApplicationTargetGroup? ServiceTargetGroup { get; private set; }
+        public AwsLogDriver? AppLogging { get; private set; }
+
+        public ScalableTaskCount? AutoScalingConfiguration { get; private set; }
+
+        public string AutoScaleTaskCount { get; } = "AutoScaleTaskCount";
+        public string AutoScaleTypeCpuType { get; } = "CpuScaling";
+        public string AutoScaleTypeRequestType { get; } = "RequestScaling";
+        public string AutoScaleTypeMemoryType { get; } = "MemoryScaling";
+
+        public string AddTargetGroup { get; } = "AddTargetGroup";
+
+        public RecipeConstruct(Construct scope, RecipeConfiguration<Configuration> recipeConfiguration, IStackProps? props = null)
```

Comment:
A recipe construct must not accept `IStackProps` in its constructor.

There should be a separate Props class like `RecipeConstructProps` which also takes care of `RecipeConfiguration<Configuration>`.

```suggestion
        public RecipeConstruct(Construct scope, string id, IRecipeConstuctProps? props = null)
```

Also having a suffix Construct doesn't go with CDK construct naming, possibly just `Recipe`?

Reply:
Removed passing in `IStackProps` but I think creating a separate props for the construct with just `RecipeConfiguration<Configuration>` which is basically props is overkill.